src/main/java/com/edmunds/tools/databricks/maven/UpsertClusterMojo.java

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright 2019 Edmunds.com, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+package com.edmunds.tools.databricks.maven;
+
+import com.edmunds.rest.databricks.DTO.LibraryDTO;
+import com.edmunds.rest.databricks.DatabricksRestException;
+import com.edmunds.rest.databricks.request.CreateClusterRequest;
+import com.edmunds.rest.databricks.service.ClusterService;
+import com.edmunds.rest.databricks.service.LibraryService;
+import com.edmunds.tools.databricks.maven.model.ClusterTemplateModel;
+import com.edmunds.tools.databricks.maven.util.ObjectMapperUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.edmunds.tools.databricks.maven.util.ClusterUtils.convertClusterNamesToIds;
+
+/**
+ * Cluster mojo, to perform databricks cluster upsert (create or update through recreation).
+ */
+@Mojo(name = "upsert-cluster")
+public class UpsertClusterMojo extends BaseDatabricksMojo {
+
+    private static final ObjectMapper OBJECT_MAPPER = ObjectMapperUtils.getObjectMapper();
+    private static final Map<String, String> SPARK_ENV_VARS = new HashMap<String, String>() {{
+        put("PYSPARK_PYTHON", "/databricks/python3/bin/python3");
+    }};
+
+    /**
+     * The databricks cluster json file that contains all of the information for how to create databricks cluster.
+     */
+    @Parameter(defaultValue = "${project.build.resources[0].directory}/databricks-plugin/databricks-cluster-settings.json", property = "dbClusterFile")
+    protected File dbClusterFile;
+
+    /**
+     * If true, command will fail if cluster with specified name already exists.
+     */
+    @Parameter(property = "failOnClusterExists")
+    protected boolean failOnClusterExists = true;
+
+    public void execute() throws MojoExecutionException {
+        ClusterTemplateModel[] cts;
+        try {
+            cts = OBJECT_MAPPER.readValue(dbClusterFile, ClusterTemplateModel[].class);
```

Comment:
I think you could use `deserialize` from `ObjectMapperUtils`

Reply:
Good point, thanks. But still ought to handle exception here.

Reply:
Right, you'll still want to handle the exception.